src/game/mod.rs

```diff
@@ -39,12 +46,24 @@ impl Plugin for InGamePlugin {
                 AnimationPlugin,
                 LevelPlugin,
                 WordMatchPlugin,
+                GameUIPlugin,
             ))
-            // Used to display words in console, remove later
-            .add_systems(Update, test_words.run_if(in_state(AppState::InGame)));
+            .add_systems(
+                Update,
+                (test_words, toggle_game_state, monitor_state).run_if(in_state(AppState::InGame)),
+            )
+            // This may not be needed
+            .add_systems(OnExit(AppState::InGame), resume_game);
     }
 }
 
+#[derive(States, Debug, Clone, Copy, Eq, PartialEq, Hash, Default)]
+pub enum InGameState {
+    #[default]
+    Running,
```

Comment:
I think its better to separate because this lets use use the schedules more appropriately. Eg we can clean up the map and enemies OnExit(InGame), and cleanup pause menu OnExit(Pause). This lets us control our systems on the plugin level instead of inside each system itself. All examples I see (except the official bevy one) seem to separate states